src/tools/input.ts

```
@@ -78,6 +80,40 @@ export const hover = defineTool({
   },
 });
 
+async function fillFormElement(
+  uid: string,
+  value: string,
+  context: McpContext,
+) {
+  const handle = await context.getElementByUid(uid);
+  try {
+    // The AXNode for an option doesn't contain its `value`. We set text content of the option as value.
+    // If the form is a combobox, we need to find the correct option by its text value.
+    // To do that, loop through the children while checking which child's text matches the requested value (requested value is actually the text content).
+    // When the correct option is found, use the element handle to get the real value.
+    const aXNode = context.getAXNodeByUid(uid);
+    if (aXNode && aXNode.role === 'combobox' && aXNode.children) {
+      for (const child of aXNode.children) {
+        if (child.role === 'option' && child.name === value && child.value) {
+          const childHandle = await child.elementHandle();
+          if (childHandle) {
+            const childValueHandle = await childHandle.getProperty('value');
+            const childValue = await childValueHandle.jsonValue();
+            if (childValue) {
+              await handle.asLocator().fill(childValue.toString());
+            }
+            break;
+          }
+        }
```

we should throw an error if no option was found